feat: isolate library & binary engine testing

Jolg42 opened the pull request:
if we can isolate tests we can then, have Slack notifications, and not me looking after it regularly 😄

Jolg42 committed:
feat: isolate library & binary engine testing

janpio commented:
You are owner of the team at https://vercel.com/teams/prisma-orm/ - so should be able to fix.
Make sure to use current credentials, we moved the deployment to a new account some time ago.

Jolg42 commented:
@janpio These errors happen after a fresh login which is concerning...

I'll clear and try again later, if we can isolate tests we can then have Slack notifications and not me looking after it regularly 😄

Jolg42 committed:
new VERCEL_WITH_REDWOOD_BINARY_PROJECT_ID env var

Jolg42 committed:
Merge remote-tracking branch 'origin/dev' into joel/isolate-binary-library

Jolg42 committed:
bump yarn to 1.22.19

because
prisma:warn Your yarn has version 1.18.0, which is outdated. Please update it to 1.19.2 or newer in order to use Prisma.

Jolg42 committed:
bump prisma to 4.3.0-dev.8

Jolg42 committed:
fix yaml for VERCEL_WITH_REDWOOD_BINARY_PROJECT_ID

Jolg42 committed:
isolate vercel-node-builder

Jolg42 committed:
isolate vercel-with-nextjs

Jolg42 committed:
isolate vercel-cli

Jolg42 committed:
isolate heroku

Jolg42 committed:
remove some deleteMany and fixed id for create

janpio commented on bundlers/parcel/src/index.js:
```diff
@@ -3,8 +3,6 @@ import { PrismaClient } from '@prisma/client'
 const client = new PrismaClient()
 
 export async function main() {
-  await client.user.deleteMany({})
```
Why remove?

janpio commented on bundlers/rollup/src/index.js:
```diff
@@ -3,8 +3,6 @@ import { PrismaClient } from '@prisma/client'
 const client = new PrismaClient()
 
 export async function main() {
-  await client.user.deleteMany({})
```
Why remove?

janpio commented on bundlers/webpack/src/index.js:
```diff
@@ -3,8 +3,6 @@ import { PrismaClient } from '@prisma/client'
 const client = new PrismaClient()
 
 export async function main() {
-  await client.user.deleteMany({})
```
Why remove?

janpio commented on databases/heroku-pgbouncer-buildpack/index.js:
```diff
@@ -14,8 +14,6 @@ const app = express()
 const port = process.env.PORT || 3000
 
 app.get('/', async (req, res) => {
-  await clientWithQueryStringParam.user.deleteMany({})
```
Why remove?

Jolg42 replied:
Because it's deleting all users, and we can make this isolated across the runs and only create / delete one user.

Same answer to the other comments

janpio replied:
Hm ok, I think this pattern is used in many other places  but might make sense to universally change.

Jolg42 replied:
Did a revert, better to do in a separate PR

janpio commented on databases/heroku-pgbouncer-buildpack/index.js:
```diff
@@ -34,18 +32,20 @@ app.get('/', async (req, res) => {
     },
   })
+  const deleteUser = await clientWithQueryStringParam.user.delete({
+    where: { id: 1 },
```
Why delete different user than created?

Jolg42 replied:
Typo, nice catch
```suggestion
    where: { id: createUser.id },
```

janpio commented on platforms-serverless-vercel/vercel-cli/api/index.js:
```diff
@@ -2,21 +2,16 @@ const { PrismaClient, Prisma } = require('@prisma/client')
 const client = new PrismaClient()
 
 export default async (req, res) => {
-  await client.user.deleteMany({})
```
Why remove?

Jolg42 committed:
Update databases/heroku-pgbouncer-buildpack/index.js

Jolg42 commented on platforms-serverless-vercel/vercel-with-redwood/api/upgrade.sh:
```diff
@@ -2,4 +2,4 @@
 set -eu
 
-yarn policies set-version 1.18.0
+yarn policies set-version 1.22.19
```
Remove this warning and it's the version used by most people (1.22)
```
prisma:warn Your yarn has version 1.18.0, which is outdated. Please update it to 1.19.2 or newer in order to use Prisma.
```

janpio replied:
What is this used/needed for? Wort adding a comment? Reviewing this I have 0 context.

Jolg42 replied:
Maybe we don't even need this anymore, I'll try removing them in a PR

janpio replied:
I honestly do not know what it was for.

Jolg42 replied:
We probably had an issue with some version of yarn at some point and we had this use this specific version workaround

Jolg42 replied:
Did a revert, turned out to break in check-for-update, added a comment
https://github.com/prisma/ecosystem-tests/commit/88e5bca3b7cd63dd7cd5e623ac3a3953274328e2

Jolg42 commented on platforms-serverless-vercel/vercel-with-redwood/api/package.json:
```diff
@@ -7,7 +7,7 @@
   "resolutions": {
-    "prisma": "3.8.0-dev.23",
-    "@prisma/client": "3.8.0-dev.23"
+    "prisma": "4.3.0-dev.8",
+    "@prisma/client": "4.3.0-dev.8"
```
That was very outdated

Jolg42 replied:
Seems we have nothing that bumps it automatically

janpio replied:
That's quite a problem then, seems this project is not working great.

janpio replied:
Best create a separate issue (if it does not exist yet)

Jolg42 replied:
I wanted to create an issue, but it seems to work mostly as expected, see https://github.com/prisma/ecosystem-tests/issues/1921#issuecomment-1223814773

The patch script works (tried it), and updates the resolutions before testing.
But we do not commit this as we only commit during `check-for-update`

Jolg42 committed:
fix expected JSON

Jolg42 committed:
Merge branch 'joel/isolate-binary-library' of github.com:prisma/e2e-tests into joel/isolate-binary-library

Jolg42 committed:
Revert "fix expected JSON"

This reverts commit 04acdd7b7245e91e9bdd01eb61ffadf31355f84c.

Jolg42 committed:
Revert "remove some deleteMany and fixed id for create"

This reverts commit 562cea594e167dd658081130a8bca8cd6e5be794.

Jolg42 committed:
rever delete user

janpio commented:
Rnadom thought: Please double check that all the jobs that you split are using the "automatic" database "provisioning" step in CI, and do not have a custom database set in the deployment platform. Otherwise this might be a bit more complicated than it looks on first sight.

Jolg42 committed:
fix vercel-node-builder

Jolg42 commented:
@janpio Let's merge this, we can do others incrementally

janpio commented:
Did you look into the databases thing?

Jolg42 commented:
@janpio the vercel deployments have a database env var set on their side. So there is nothing else to do here I think.

I think to avoid the constraints issues, we can drop `@unique` on email and remove the fixed id used in the tests. (= still use 1 database for both)

janpio commented on platforms-serverless-vercel/vercel-with-redwood/package.json:
```diff
@@ -14,8 +14,8 @@
   "resolutions": {
-    "prisma": "3.8.0-dev.23",
-    "@prisma/client": "3.8.0-dev.23"
+    "prisma": "4.3.0-dev.8",
+    "@prisma/client": "4.3.0-dev.8"
```
Probably also worth an issue if this is relevant.

janpio commented on platforms-serverless-vercel/vercel-with-redwood/run.sh:
```diff
@@ -2,25 +2,31 @@
 set -eu
 
-yarn policies set-version 1.18.0
+yarn policies set-version 1.22.19
```
Also comment

janpio commented on platforms-serverless-vercel/vercel-with-redwood/upgrade.sh:
```diff
@@ -2,4 +2,4 @@
 set -eu
 
-yarn policies set-version 1.18.0
+yarn policies set-version 1.22.19
```
comment

janpio commented on platforms-serverless-vercel/vercel-with-redwood/web/upgrade.sh:
```diff
@@ -2,4 +2,4 @@
 set -eu
 
-yarn policies set-version 1.18.0
+yarn policies set-version 1.22.19
```
comment

janpio commented:
> @janpio the vercel deployments have a database env var set on their side. So there is nothing else to do here I think.

So I assume the new deployment just got the same database set as the old one now?
It would be better if we could "inject" that env var from the script and make them use the same, auto generated database as most other tests use. (But not necessary for merge here, but please issue)
